python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+    )
+    astrometry = pexConfig.ConfigurableField(
+        target = AstrometryTask,
+        doc = "Task to load and match reference objects. Only used if measurePsf can use matches. "
+            "Warning: matching will only work well if the initial WCS is accurate enough "
+            "to give good matches (roughly: good to 3 arcsec across the CCD).",
+    )
+    measurePsf = pexConfig.ConfigurableField(
+        target = MeasurePsfTask,
+        doc = "Measure PSF",
+    )
+    repair = pexConfig.ConfigurableField(
+        target = RepairTask,
+        doc = "Remove cosmic rays",
+    )
+
+    def setDefaults(self):
+        pexConfig.Config.setDefaults(self)
+        # just detect bright stars; includeThresholdMultipler=10 seems large,
+        # but these are the values we have been using
+        self.detectAndMeasure.detection.thresholdValue = 5.0
+        self.detectAndMeasure.detection.includeThresholdMultiplier = 10.0
+        # do not deblend, as it makes a mess
+        self.detectAndMeasure.doDeblend = False
+        # measure and apply aperture correction; note: measuring and applying aperture
+        # correction are disabled until the final measurement, after PSF is measured
+        self.detectAndMeasure.doMeasureApCorr = True
+        self.detectAndMeasure.measurement.doApplyApCorr = "yes"
+        # minimal set of measurements needed to determine PSF
+        self.detectAndMeasure.measurement.plugins.names = [
+            "base_PixelFlags",
+            "base_SdssCentroid",
+            "base_SdssShape",
+            "base_GaussianFlux",
+            "base_PsfFlux",
+            "base_CircularApertureFlux",
+        ]
+
+    def validate(self):
+        if self.detectAndMeasure.doMeasureApCorr and not self.measurePsf:
+            raise RuntimeError("Must measure PSF to measure aperture correction, "
+                "because flags determined by PSF measurement are used to identify "
+                "sources used to measure aperture correction")
+
+## \addtogroup LSST_task_documentation
+## \{
+## \page CharacterizeImageTask
+## \ref CharacterizeImageTask_ "CharacterizeImageTask"
+## \copybrief CharacterizeImageTask
+## \}
+
+class CharacterizeImageTask(pipeBase.CmdLineTask):
+    """!Measure bright sources and use this to estimate background and PSF of an exposure
+
+    @anchor CharacterizeImageTask_
+    
+    @section pipe_tasks_characterizeImage_Contents  Contents
+
+     - @ref pipe_tasks_characterizeImage_Purpose
+     - @ref pipe_tasks_characterizeImage_Initialize
+     - @ref pipe_tasks_characterizeImage_IO
+     - @ref pipe_tasks_characterizeImage_Config
+     - @ref pipe_tasks_characterizeImage_Debug
+     - @ref pipe_tasks_characterizeImage_Example
+
+    @section pipe_tasks_characterizeImage_Purpose  Description
+
+    Given an exposure with defects repaired (masked and interpolated over, e.g. as output by IsrTask):
+    - detect and measure bright sources
+    - repair cosmic rays
```

Comment:
This seems to neglect repairing defects.  There's a reason both of those are grouped in the `RepairTask` --- both (should) depend on the PSF (the interpolation doesn't currently because the implementation uses a hard-coded PSF), and so both should be run with the final PSF.